### PR TITLE
planner: disable projection elimination for select plan of update

### PR DESCRIPTION
**Pull request by eurekaka**
### What problem does this PR solve? 

Before this PR:
```
mysql> create table t(a int, b int);
Query OK, 0 rows affected (0.01 sec)

mysql> explain update t t1, (select distinct b from t) t2 set t1.b = t2.b;
ERROR 1105 (HY000): Can't find column test.t2.b in schema Column: [test.t1.a,test.t1.b,test.t1._tidb_rowid,test.t.b] Unique key: []
```

### What is changed and how it works?

The root cause of the error is that during projection elimination, we do not resolve columns in `OrderedList` field of `Update`, so this PR avoid this by reset the flagProjectionElimination when building select plan for update.

### Check List 

Tests 

 - Unit test

Code changes

N/A

Side effects

N/A

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: planner/core/logical_plan_builder.go [hunk 1]
```diff
@@ -2585,7 +2585,9 @@ func (b *PlanBuilder) buildUpdate(update *ast.UpdateStmt) (Plan, error) {
 
 	updt := Update{OrderedList: orderedList}.Init(b.ctx)
 	updt.SetSchema(p.Schema())
-	updt.SelectPlan, err = DoOptimize(b.optFlag, p)
+	// We cannot apply projection elimination when building the subplan, because
+	// columns in orderedList cannot be resolved.
+	updt.SelectPlan, err = DoOptimize(b.optFlag&^flagEliminateProjection, p)
```
Comment (zz-jason) [inline planner/core/logical_plan_builder.go]: I'm not sure whether there will be some bugs if we disable the projection elimination rule. As you mentioned in the PR description that "The root cause of the error is that during projection elimination, we do not resolve columns in `OrderedList` field of `Update`", can we fix this issue in the projection elimination rule for the `Update` operator?

**Comment by XuHuaiyu** [comment]
Can we set Update.Schema to the SelectPlan.Schema after SelectPlan is optimized?

File: planner/core/logical_plan_builder.go (hunk 1 above)
Comment (eurekaka) [reply planner/core/logical_plan_builder.go]: Then we need to pass an additional parameter of type `Assignment` to `DoOptimize`, it is pretty ugly and ad-hoc IMHO.

**Comment by eurekaka** [comment]
@XuHuaiyu The error is raised because column in `OrderedList` of `Update` cannot find match in schema of `SelectPlan`, so I think setting schema of `Update` would not help.

**Pull request merged**